### PR TITLE
[Function] Add compare_version function and ut

### PR DESCRIPTION
**Pull request by carlvinhust2012**
# Proposed changes
Add compare_version function and ut. the function will return the compared result of two version.
Issue Number: close #xxx

## Problem Summary:

Describe the overview of changes.
1.Add compare_version function and ut.
2.add md files for this compare_version function.

## Checklist(Required)

1. Does it affect the original behavior: No
2. Has unit tests been added: Yes
3. Has document been added or modified: Yes
4. Does it need to update dependencies:No
5. Are there any changes that cannot be rolled back: No

## Further comments
If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exprs/utility_functions.cpp [hunk 1]
```diff
@@ -39,4 +39,82 @@ BooleanVal UtilityFunctions::sleep(FunctionContext* ctx, const IntVal& seconds)
     return BooleanVal(true);
 }
 
+static void convert_to_string(doris_udf::StringVal& src, std::string& str_dst) {
```
Comment (yangzhg) [inline be/src/exprs/utility_functions.cpp:42]: why called `convert_to_string` ？ just a concat ?

File: be/src/exprs/utility_functions.cpp [hunk 2]
```diff
@@ -39,4 +39,82 @@ BooleanVal UtilityFunctions::sleep(FunctionContext* ctx, const IntVal& seconds)
...
+    for (int i = 0; i < src.len; i++) {
+        str_dst += src.ptr[i];
+    }
+    return;
+}
+
+static BooleanVal split_string(std::string& str_src, std::vector<std::string>& str_vec_dst) {
```
Comment (yangzhg) [inline be/src/exprs/utility_functions.cpp:49]: there is already have split in the util, refer to be/src/util/string_util.h or gutils

**Review by yangzhg — COMMENTED** [review]
Why not `compare_version(ver1 , ver2)`  return 0 if equal, 1 if ver1 > ver2, -1 if ver1 < ver2 ？
In addition, there is still a lot of room for optimization in the implementation of the program

File: be/src/exprs/utility_functions.cpp (hunk 1 above)
Comment (carlvinhust2012) [reply be/src/exprs/utility_functions.cpp:42]: just want to convert the type from 'StringVal' to 'string'.

File: be/src/exprs/utility_functions.cpp (hunk 2 above)
Comment (carlvinhust2012) [reply be/src/exprs/utility_functions.cpp:49]: I will re-check this split function.

**Comment by carlvinhust2012** [comment]
> Why not `compare_version(ver1 , ver2)` return 0 if equal, 1 if ver1 > ver2, -1 if ver1 < ver2 ？ In addition, there is still a lot of room for optimization in the implementation of the program

**Comment by carlvinhust2012** [comment]
> Why not `compare_version(ver1 , ver2)` return 0 if equal, 1 if ver1 > ver2, -1 if ver1 < ver2 ？ In addition, there is still a lot of room for optimization in the implementation of the program

We discussed this approach with the business, but they still prefer to use the current approach.

**Comment by carlvinhust2012** [comment]
> > Why not `compare_version(ver1 , ver2)` return 0 if equal, 1 if ver1 > ver2, -1 if ver1 < ver2 ？ In addition, there is still a lot of room for optimization in the implementation of the program

We discussed this approach with the business, but they still prefer to use the current approach.

**Comment by carlvinhust2012** [comment]
the customized versions have been provided to customer, so this pr could be closed.

**Pull request closed without merging**